database/generate-records.js

```diff
@@ -63,34 +91,52 @@ const writeNRecords = (n, document, callback) => {
   const options = {
     verticalOutput: false,
     mapHeaders: (header) => '',
-    includeHeaders: false,
+    // includeHeaders: false,
   }
+  throughSheet.write('itemsId,imagesId\n');
+  // noSQLSheet.write('id,name,description,materials,sustainibility,packaging,sizes,imageUrls\n');
   const write = async () => {
-    let ok = true;
-    while (i > 0 && ok) {
+    let ok1 = true;
+    let ok2 = true;
+    while (i > 0 && ok1 && ok2) {
       i -= 1;
       id += 1;
-      let record = await jsonexport(generateRecord(id), options);
+      let rawRecord = generateRecord(id);
+      let record = await jsonexport(rawRecord, options);
+      let intermediateRecord = includeImageUrls(rawRecord);
+      let noSQLRecord = await jsonexport(intermediateRecord, {verticalOutput: false});
       if (id !== 1) {
         record = record.split('\n').slice(1).join('');
+        noSQLRecord = noSQLRecord.split('\n').slice(1).join('');
       }
       record += '\n';
+      noSQLRecord += '\n';
       if (i === 0) {
         document.write(record, 'utf8', () => {
-          console.log(`${id} records written in ${new Date() - startTime} ms`);
-          callback(new Date());
+          noSQLSheet.write(noSQLRecord, 'utf8', () => {
+            throughSheet.write(getImageRelations(id), 'utf8', () => {
+              console.log(`${id} records written in ${new Date() - startTime} ms`);
+              callback(new Date());
+            });
+          });
         });
       } else {
-        ok = document.write(record, 'utf8');
+        ok1 = document.write(record, 'utf8');
+        ok2 = noSQLSheet.write(noSQLRecord, 'utf8');
+        throughSheet.write(getImageRelations(id), 'utf8');
       }
     }
-    if (i > 0) {
+    if (i > 0 && !ok1) {
       document.once('drain', write);
     }
+    if (i > 0 && !ok2) {
+      noSQLSheet.once('drain', write);
+    }
   }
   write();
 };
 
 module.exports = (n, callback) => {
-  writeNRecords(n, writeRecords, callback);
+  writeImageRecords(imageDataSheet);
```

Comment:
What is the difference between throughSheet and noSQLSheet?

Reply:
yeah, to be honest my naming convention is pretty bad here. throughSheet is one of three tables for PostgreSQL--it has 2 columns to relate image ids to product ids. noSQLSheet is the one and only table for Cassandra. It practically stores the entire object that I want returned on the client side.